hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,343 @@
+#!/bin/bash
+
+conf_file=""
+
+declare -a netnses
+
+declare -A pfs
+declare -A pcis
+declare -A pf_port_names
+declare -A pf_switch_ids
+
+TIMEOUT="${TIMEOUT:-2}"
+POLL_INTERVAL="1"
+
+while test $# -gt 0; do
+  case "$1" in
+
+   --netns | -n)
+      netnses+=("$2")
+      shift
+      shift
+      ;;
+
+   --pf | -d)
+      if [[ -z "${netnses[-1]}" ]];then
+          echo "ERROR: No netns is specified, please provide one using the --netns before using --pf."
+          exit 1
+      fi
+
+      pfs["${netnses[-1]}"]+="$2 "
+      shift
+      shift
+      ;;
+
+   --conf-file | -c)
+      conf_file=$2
+      shift
+      shift
+      ;;
+
+   --help | -h)
+      echo "
+vf-netns-switcher.sh --netns <> --pf <> [--conf-file <>]:
+
+	--netns | -n		Netns to switch the interface PFs and VFs to. This needs to be provided before the pfs
+				to associate the pfs to the netns. This flag can be repeated.
+
+	--pf | -d		The PF to switch it and its VFs to the specified netns. This must be provided after a
+				netns to associate the pf to the last provided netns. This option can be repeated.
+
+	--conf-file | -c	A file to read confs from, this will override cli flags. Conf file should be of the form:
+				- netns: <netns1>
+				  pfs:
+				  - <pf1>
+				  - <pf2>
+				- netns: <netns2>
+				  pfs:
+				  - <pf3>
+				  - <pf4>
+
+"
+      exit 0
+      ;;
+
+   *)
+      echo "No such option!!"
+      echo "Exitting ...."
+      exit 1
+  esac
+done
+
+get_pcis_from_pfs(){
+    local worker_netns="${1}"
+    shift
+    local interfaces="$@"
+    for interface in $interfaces; do
+        pcis["$interface"]="$(get_pci_from_net_name "$interface" "$worker_netns")"
+    done
+}
+
+get_pci_from_net_name(){
+    local interface_name=$1
+    local worker_netns="${2:-$netns}"
+
+    if [[ -z "$(ip l show $interface_name)" ]];then
+        if [[ -n "$(docker exec -t ${worker_netns} ip l show $interface_name)" ]];then
+            ip netns exec ${worker_netns} bash -c "basename \$(readlink /sys/class/net/${interface_name}/device)"
+            return 0
+        fi
+        echo ""
```

Comment:
did you mean to log an error here ?

Reply:
no, it is used as a return value for the function. like `return null`.